rs/pocket_ic_server/src/state_api/state.rs

```diff
@@ -1103,11 +1108,11 @@ impl ApiState {
             let (tx, mut rx) = mpsc::channel::<()>(1);
             let handle = spawn(async move {
                 debug!("Starting auto progress for instance {}.", instance_id);
-                let mut now = Instant::now();
+                let mut now = SystemTime::now();
```

Comment:
[Instant](https://doc.rust-lang.org/std/time/struct.SystemTime.html#method.duration_since) has the advantage of being non-decreasing, which makes the unwrap_or_default below unnecessary. (Tangent: Does AdvanceTimeAndTick actually respect an argument of 0 or is the 1ns minimum also applicable there?). So what is the advantage of SystemTime, if we are interested in time differences only anyway? 

I don't quite understand 
>so that time advances properly also if the server's machine is hibernated and Instant does not advance.

Reply:
At the moment, if you hibernate your laptop with a live PocketIC instance and later turn it on again, the PocketIC time will lag behind because Instant won't increase while your laptop was hibernated.

Reply:
Aha. The time spent sleeping is not accounted for. 
And `set_time(now)` does not solve the problem because we would end up setting time into the past during certain times of the year. 

Still, the given implementation gives a wrong reading during a whole hour after the summertime switch, right? 

Wall Time --- PocketIC Time
02:59   ---   02:59
>time changes from 3 to 2. 
02:00   ---   **02:59**
02:01   ---   **02:59**
...
03:00 --- 03:00

Reply:
> And set_time(now) does not solve the problem

This wouldn't work if we want to allow advancing time of a live instance into the future: with set_time(now), the live mode will try to reset the time back.

The summer time switch does not matter because systemtime returns UTC.

Reply:
>The summer time switch does not matter because systemtime returns UTC.

Ah yes, so the whole-hour difference situation is not an issue. 

But it [explicitly warns](https://doc.rust-lang.org/std/time/struct.SystemTime.html): 

>Distinct from the [Instant](https://doc.rust-lang.org/std/time/struct.Instant.html) type, this time measurement is not monotonic.

I guess this concerns the leap seconds only and we are fine since that amount of time dilation is fine.